handlers/geolocation/geolocation.go

```diff
@@ -150,31 +160,43 @@ func (c *GeolocationHandlersCollection) HandleStreamSource(ctx context.Context,
 
 	latStr := fmt.Sprintf("%f", lat)
 	lonStr := fmt.Sprintf("%f", lon)
-	dtscURL, err := c.Balancer.MistUtilLoadSource(context.Background(), payload.StreamName, latStr, lonStr)
-	if err != nil {
-		glog.Errorf("error querying mist for STREAM_SOURCE: %s", err)
-
-		playbackID := payload.StreamName
-		parts := strings.Split(playbackID, "+")
-		if len(parts) == 2 {
-			playbackID = parts[1] // take the playbackID after the prefix e.g. 'video+'
+	for i := 0; i < streamSourceRetries; i++ {
+		dtscURL, err := c.Balancer.MistUtilLoadSource(context.Background(), payload.StreamName, latStr, lonStr)
+		if err == nil {
+			return c.resolveReplicatedStream(dtscURL, payload.StreamName)
 		}
-		pullURL, err := c.getStreamPull(playbackID)
-		if err != nil {
+
+		glog.Errorf("error querying mist for STREAM_SOURCE: %s", err)
+		pullURL, err := c.getStreamPull(playbackID(payload.StreamName))
+		if err != nil && !errors.Is(err, errLockPull) {
 			glog.Errorf("getStreamPull failed for %s: %s", payload.StreamName, err)
+			return "push://", nil
 		} else if pullURL != "" {
 			glog.Infof("replying to Mist STREAM_SOURCE with stream pull request=%s response=%s", payload.StreamName, pullURL)
 			return pullURL, nil
 		}
+		// another Mist node is currently pulling the stream, sleep and retry
+		time.Sleep(streamSourceRetryInterval)
```

Comment:
Might be useful to have a log line here at least in the beginning for debugging

Reply:
Good point, updated.